code/modules/clothing/neck/horrific_necktie.dm

```diff
@@ -0,0 +1,105 @@
+/obj/item/clothing/neck/tie/disco
+	name = "horrific necktie"
+	icon = 'icons/obj/clothing/neck.dmi'
+	icon_state = "eldritch_tie"
+	post_init_icon_state = null
+	desc = "The necktie is adorned with a garish pattern. It's disturbingly vivid. Somehow you feel as if it would be wrong to ever take it off. It's your friend now. You will betray it if you change it for some boring scarf."
+	clip_on = TRUE
+	greyscale_config = null
+	greyscale_config_worn = null
+	greyscale_colors = null
+
+	/// Are we grabbing a spirit?
+	var/using = FALSE
+	///The mob that inhabits us, once posessed.
+	var/mob/living/basic/tie/possessed
+
+/obj/item/clothing/neck/tie/disco/Destroy()
+	QDEL_NULL(possessed)
+	return ..()
+
+/obj/item/clothing/neck/tie/disco/examine(mob/user)
+	. = ..()
+	if(isnull(possessed))
+		. += span_notice("It may be given sentience by [EXAMINE_HINT("using it in hand")].")
+
+/obj/item/clothing/neck/tie/disco/attack_self(mob/living/user, modifiers)
+	if(using || !isnull(possessed))
+		return ..()
+	if(!(GLOB.ghost_role_flags & GHOSTROLE_STATION_SENTIENCE))
+		user.balloon_alert(user, "spirits are unwilling!")
+		return ..()
+
+	using = TRUE
+	to_chat(user, span_notice("You plumb the depths of your Inland Empire. Whispers seem to emaninate from [src], as though it had somehow come to life; could it be?"))
+
+	var/mob/speaking_tie = SSpolling.poll_ghosts_for_target(
+		question = "Do you want to play as the spirit of [span_danger("[user.real_name]'s")] [span_notice("horrific necktie")]?",
+		check_jobban = ROLE_PAI,
+		poll_time = 20 SECONDS,
+		checked_target = user,
+		ignore_category = POLL_IGNORE_HORRIFIC_NECKTIE,
+		alert_pic = user,
+		role_name_text = "horrific necktie",
+	)
+	if(QDELETED(src))
+		return
+	if(speaking_tie)
+		possessed = new(src, user)
+		possessed.PossessByPlayer(speaking_tie.ckey)
+	else
+		to_chat(user, span_warning("The whispers coming from [src] fade and are silent again... Was it all your imagination? Maybe you can try again later."))
+	using = FALSE
+
+///The mob that inhabits the tie when posessed.
+/mob/living/basic/tie
+	name = "horrific necktie"
+	gender = NEUTER
+	mob_biotypes = MOB_SPIRIT
+	faction = list()
+	unsuitable_cold_damage = 0
+	unsuitable_heat_damage = 0
+	unsuitable_atmos_damage = 0
+
+	///The only person who can hear us is the one who activated us, set by the tie item.
+	var/mob/living/hears_us
+	///Innate ability for the tie to change its name constantly, in case they want to play several voices at once.
+	var/datum/action/innate/change_name/name_change
```

Comment:
You don't need to hold a reference to the action, the refs are already cleared on deletion even without you doing that.